components/permissions/permission_origin_lifetime_monitor.h

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_PERMISSIONS_PERMISSION_ORIGIN_LIFETIME_MONITOR_H_
+#define BRAVE_COMPONENTS_PERMISSIONS_PERMISSION_ORIGIN_LIFETIME_MONITOR_H_
+
+#include <string>
+
+#include "base/callback.h"
+#include "url/gurl.h"
+
+namespace permissions {
+
+// Helper to support an origin-based permission lifetime logic.
+class PermissionOriginLifetimeMonitor {
+ public:
+  virtual ~PermissionOriginLifetimeMonitor() = default;
+
+  // Set a callback to call when a permission origin is destroyed.
+  // Callback will receive a string returned by
+  // |SubscribeToPermissionOriginDestruction|.
+  virtual void SetOnPermissionOriginDestroyedCallback(
+      base::RepeatingCallback<void(const std::string&)> callback) = 0;
+
+  // Subscribe to a permission origin destruction. Returns a string key which
+  // will be used in a callback. An empty key can be returned if a permission
+  // origin can not be observed.
+  virtual std::string SubscribeToPermissionOriginDestruction(
```

Comment:
I'd like to have `Ephemeral` in this name, but can't really construct one. `SubscribeToEphemeralOriginDestruction` ?

Reply:
there is deliberately no any 'ephemeral' mentions in this interface and in ```PermissionLifetimeManager```. it's just about observing an arbitrary origin destruction. I don't think we need to introduce 'ephemeral' term on this level currently.
it might be clearer to add a ```component``` with ```ephemeral``` word in its naming that can track ```TLDEphemeralLifetime``` destruction so it can be used by other components and inside ```chrome``` itself, but I'm not sure if it's worth doing it in the current task.